src/reactions/halloween.ts

```diff
@@ -0,0 +1,34 @@
+import { logger } from '../utilities/logger';
+import { ReactionDef } from './reaction-def';
+
+export const halloweenReaction: ReactionDef = {
+  emoji: '🎃',
+  description: 'Adds the halloween role to the user who reacts',
+  command: async (reaction, { config, user }) => {
+    const hallowRole = reaction.message.guild?.roles.cache.find(
+      (role) => role.name === config.HALLOWEEN_ROLE
+    );
+    if (!hallowRole) {
+      logger.warn('Halloween role not found');
+      return;
+    }
+    if (reaction.message.id !== config.HALLOWEEN_ANNOUNCEMENT) {
+      return;
+    }
+    if (user.partial) {
+      user = await user.fetch();
+    }
+    const target = reaction.message.guild?.member(user);
+    if (!target) {
+      logger.warn('user error');
```

Comment:
can we add a more descriptive warning message?

Reply:
Filled it out a bit - this *shouldn't* actually trigger, tho. 
TS expects that `guild` might be undefined (which is the case for DMs) - including this check prevents compile errors in the later role assignment. 
But since it's locked to the specific message ID anyway, the command should never get this far in a DM.